File: T/2032-two-out-of-three/main.go
package main

import "fmt"

func twoOutOfThree(nums1 []int, nums2 []int, nums3 []int) []int {
	m1, m2, m3 := map[int]int{}, map[int]int{}, map[int]int{}
	for _, v := range nums1 {
		m1[v] = 1
	}
	for _, v := range nums2 {
		m2[v] = 1
	}
	for _, v := range nums3 {
		m3[v] = 1
	}
	total := map[int]int{}
	for k := range m1 {
		total[k]++
	}
	for k := range m2 {
		total[k]++
	}
	for k := range m3 {
		total[k]++
	}
	ret := []int{}
	for k, v := range total {
		if v >= 2 {
			ret = append(ret, k)
		}
	}
	return ret
}

func main() {
	fmt.Println()
}

File: T/2032-two-out-of-three/main.rs
struct Solution {}

use std::collections::HashMap;
impl Solution {
  pub fn two_out_of_three(nums1: Vec<i32>, nums2: Vec<i32>, nums3: Vec<i32>) -> Vec<i32> {
    let mut m: HashMap<i32, i32> = HashMap::new();

    for n in nums1 {
      let mut v = m.entry(n).or_insert(0);
      *v |= 1;
    }

    for n in nums2 {
      let mut v = m.entry(n).or_insert(0);
      *v |= 2;
    }

    for n in nums3 {
      let mut v = m.entry(n).or_insert(0);
      *v |= 4;
    }
    let mut ret: Vec<i32> = Vec::new();
    for (k, v) in m {
      if v.count_ones() > 1 {
        ret.push(k);
      }
    }
    ret
  }
}
